docs/WritingTests.asciidoc

```diff
@@ -28,38 +28,10 @@ is already familiar with Perl.
 
 == API
 [id="api"]
+:testapi: https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm[os-autoinst]
 
-https://github.com/os-autoinst/os-autoinst[os-autoinst] provides the
-API for the tests. The most commonly used functions are:
-
-* +get_var NAME[, DEFAULT]+ Returns the content of a variable if available; otherwise returns the specified default value or undef.
-* +check_var NAME, CONTENT+ Check content of a variable, treating undef as empty.
-* +send_key KEY[, WAITIDLE]+ Send key to openQA instance. Call +wait_idle+ at the end if WAITIDLE is set. (e.g. +send_key "alt-o", 1+)
-* +type_string STRING+ Typing the given string. (e.g. +type_string "yes\n"+)
-* +save_screenshot+ Capture and save a screenshot in the test results.
-* +check_screen NEEDLE[, TIMEOUT]+ Wait until NEEDLE is seen on the screen. Return undef if not found within TIMEOUT (default 30s).
-* +assert_screen NEEDLE[, TIMEOUT]+ Wait until NEEDLE is seen on the screen. Abort test and mark as failed if not found within TIMEOUT (default 30s).
-* +record_soft_failure+ Record a workaround applied in the test.
-* +wait_idle TIMEOUT+  Wait for max TIMEOUT seconds until system is idle.
-* +select_console STRING+ Switch to another TTY (text based or graphical), console names are defined by the distribution.
-* +wait_serial REGEX[, TIMEOUT]+ Wait until REGEX is seen on serial port. Return 0 if not found after TIMEOUT (default 90s)
-* +upload_logs FILE+ Type in necessary shell commands to save FILE in the test results
-* +assert_and_click NEEDLE[, BUTTON]+ Wait until NEEDLE is seen on the screen and click it in the middle of the smallest defined area
-* +wait_still_screen [TIME]+ Wait for the screen to stop changing for TIME seconds
-* +mouse_hide+ Moves the mouse cursor into the edge of the screen to hide it within needles
-* +script_output SCRIPT[, TIMEOUT]+ Returns the text output of the given bash script. The script is called with bash -xe and STDOUT is returned
-* +validate_script_output SCRIPT, CODE+ Checks the text output of the bash script using the given callback (e.g. +validate_script_output "ls -la $HOME", sub { m/.xsession-errors/ }+)
-* +assert_shutdown [TIMEOUT]+ Wait for max TIMEOUT seconds for system to shut down (default 30s)
-
-The following API is distribution specific and subject to change
```

Comment:
Obviously this needs to be somehow clarified or added to another doc, or perhaps here... about how it can be distribution specific and/or how to overload...